rfcXXX-github-issues-workflow.md

```diff
@@ -0,0 +1,90 @@
+---
+RFC:
+Author: Serdar Sutay <serdar@getchef.com>
+Status: Draft
+Type: Process
+---
+
+# Chef GitHub Issues Workflow
+
+This RFC codifies the workflow Chef uses to track bug reports and contributions using [GitHub Issues](https://github.com/opscode/chef/issues). Goals of this workflow are:
+
+## Terms
+
+In the context of this document below terms are used:
+
+* **Chef Users:** End users of Chef who participate in this workflow by filing **issues**.
+* **Contributors:** Users of Chef who would like to improve Chef. They participate in this workflow by submitting **contributions** to Chef.
+* **Chef Maintainers:** Developers of Chef who are responsible for one or more **subcomponents**.
+* **Lieutenants:** (Lt.) Developers of Chef who lead the maintenance of a subcomponent.
+* **Issue**: A filed **Github Issue** which can be a bug report, feature request, style change request.
+* **Contribution**: A filed **Github Pull Request** (PR).
+* **Subcomponents**: A sub-area of Chef project which is lead by a **Lt.** and a set of **maintainers**.
+
+## Goals
+
+Chef Users can:
+
+* Be aware of the cause & the next steps for the issues they have filed.
+
+Contributors can:
+
+* Identify the state of their contribution.
+
+Chef Maintainers can:
+
+* See the list of contributions they need to review in a single page.
+* See the list of issues they need to investigate in a single page.
+
+
+## Workflow
+
+![Chef GitHub Issues Workflow](./GithubIssuesWorkflow.png)
+
+
+## Notes
+
+### Common Steps for Issues & PRs
+
+The first step of Chef Github Issues Workflow is to identify the subcomponent for which the issue or PR belongs to. Once the subcomponent is identified, the rest of the workflow is the responsibility of the maintainers of the identified subcomponent.
+
+### Issues Workflow
+
+The issues workflow can be broken down into these high level steps:
+
+1. Understand the issue and determine the resolution.
+1. Identify the major version the issue should be fixed in.
+1. Label the issue to indicate completeness of **Triage** process.
+
+After this point any fix for an issue should follow the contribution workflow.
+
+#### Rules
+
+* If an issue is not commented on for more than a year, the issue is retired by closing.
+
+### Contribution Workflow
+
+The contribution workflow can be broken down into these high level steps:
+
+1. Maintainers check the contribution and give **:+1:** if the code is looking good, or leave a comment that clearly identifies what is needed for the contribution to move forward.
+1. If the simple majority of maintainers **:+1:** a PR, it is marked with `Ready to Merge` label.
+1. PRs marked with `Ready to Merge` will be merged by maintainers of the subcomponent after including the required documentation updates.
```

Comment:
As "merge monkeys" for Chef we've tried to rebase and potenially squash/fixup branches before merging them, (don't use the github green merge button) so the git history isn't 50% merge commits.

It's generally preferred that CHANGELOG updates not be included in pull requests because that makes almost every merge have a conflict that needs to be resolved. It's easier to encourage good commit messages and make a CHANGELOG update on merge. DOC_CHANGES and RELEASE_NOTES are sometimes encouraged to be completed by the contributor because they're more familiar with the change and it's rarely a one-liner when required.

It would be best to add words here somewhere regarding the value of rebasing, squashing/fixuping junk commits, good commit messages, etc.